client/state/selectors/is-site-on-paid-plan.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * Internal dependencies
+ */
+import { getCurrentPlan } from 'state/sites/plans/selectors';
+import { PLAN_FREE } from 'lib/plans/constants';
+
+/**
+ * Returns true if site is on a paid plan, false if the site is not,
+ * or null if the site is unknown.
+ *
+ * @param {Object} state Global state tree
+ * @param {Number} siteId Site ID
+ * @return {?Boolean} Whether site is on a paid plan
+ */
+const isSiteOnPaidPlan = ( state, siteId ) => {
+	const currentPlan = getCurrentPlan( state, siteId );
+
+	if ( ! currentPlan ) {
+		return null;
```

Comment:
Nitpick, but I wonder if we really need to know the `null` case? (I'm actually curious when that could happen) It forces the user of this selector to use the hacky `!!` operator (and to know that they should it). Based on the name, I'd say returning only boolean values is safer and more intuitive.